File: bubbleSort.py
def bubble_sort(my_list):
    n = len(my_list)
    # Perform n - 1 passes
    for i in range(n - 1):
        # Initialize flag to False for each pass
        flag = False
        # In each pass, compare each element with its adjacent element
        for j in range(n - i - 1):
            if my_list[j] > my_list[j + 1]:
                # Swap the elements
                my_list[j], my_list[j + 1] = my_list[j + 1], my_list[j]
                # Set flag to True if a swap occurs
                flag = True
        # If no swaps occur in a pass, the my_list is already sorted, so break the loop
        if not flag:
            break

my_list = [8, 8, 3, 5, 4]
bubble_sort(my_list)
print("Sorted array:", my_list)

File: bubbleSort.java
public class BubbleSort {
    public static void bubbleSort(int[] A) {
        int n = A.length;
        // Perform n - 1 passes
        for (int i = 0; i < n - 1; i++) {
            // In each pass, one element is compared with the next element
            for (int j = 0; j < n - i - 1; j++) {
                if (A[j] > A[j + 1]) {
                    // Swap the elements
                    int temp = A[j];
                    A[j] = A[j + 1];
                    A[j + 1] = temp;
                }
            }
        }
    }

    public static void main(String[] args) {
        int[] arr = {64, 34, 25, 12, 22, 11, 90};
        bubbleSort(arr);
        System.out.print("Sorted array: ");
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
    }
}
